core/src/bank_forks.rs

```diff
@@ -256,22 +261,26 @@ impl BankForks {
             .expect("root must exist in BankForks");
 
         let mut add_snapshot_time = Measure::start("add-snapshot-ms");
-        snapshot_utils::add_snapshot(&config.snapshot_path, &bank, slots_since_snapshot)?;
+        snapshot_utils::add_snapshot(&config.snapshot_path, &bank)?;
         add_snapshot_time.stop();
         inc_new_counter_info!("add-snapshot-ms", add_snapshot_time.as_ms() as usize);
 
         // Package the relevant snapshots
         let slot_snapshot_paths = snapshot_utils::get_snapshot_paths(&config.snapshot_path);
-
-        // We only care about the last MAX_CACHE_ENTRIES snapshots of roots because
-        // the status cache of anything older is thrown away by the bank in
-        // status_cache.prune_roots()
-        let start = slot_snapshot_paths.len().saturating_sub(MAX_CACHE_ENTRIES);
+        let latest_slot_snapshot_paths = slot_snapshot_paths.last().ok_or_else(|| {
```

Comment:
You can call unwrap() here. We just called add_snapshot above, so there better be at least one snapshot

Reply:
I think the less crashy the better. It's more verbose this way too.

Reply:
call .expect(<Message>) to bee more verbose. This is a correctness assertion, the crash should never happen and helps reason about the code